src/Surfnet/StepupMiddleware/CommandHandlingBundle/Tests/EventHandling/BufferedEventBusTest.php

```diff
@@ -85,38 +85,44 @@ public function flushing_succesfully_empties_the_buffer_to_prevent_flushing_the_
     }
 
     /**
-     * This is tested by publishing an event when flushing and flushing again afterwards
-     *
      * @test
      * @group event-handling
      */
-    public function new_event_can_be_buffered_while_flushing()
+    public function an_event_caused_by_an_event_in_the_current_buffer_being_flushed_is_buffered_and_flushed_after_events_in_the_current_buffer()
     {
-        $event = $this->createDummyDomainMessage();
         $bus = new BufferedEventBus();
 
-        // in php7 replace this with anonymous class
-        $listener = new OnFirstCallPublishesToBusAndCountingCallsEventListener(
-            $bus,
-            new DomainEventStream([$event])
+        $firstEventInCurrentBuffer = $this->createDummyDomainMessage(null, 'First event in current buffer');
+        $eventCausedByFirstEvent = $this->createDummyDomainMessage(
+            null,
+            'Event caused by first event in current buffer'
         );
+        $secondEventInCurrentBuffer = $this->createDummyDomainMessage(null, 'Second event in current buffer');
 
-        $bus->subscribe($listener);
+        $expectedEventSequence = [$firstEventInCurrentBuffer, $secondEventInCurrentBuffer, $eventCausedByFirstEvent];
```

Comment:
Seems this should be lower in the method, say above `$actualEventSequence`, so that it is easier to grok what is going on. Would also recommend creating the `$eventCausedByFirstEvent` after creating the two events that are published directly.